e2e/utils/test-utils.ts

```diff
@@ -306,3 +310,6 @@ export function isValidDate(date: string) {
   }
   return d.toISOString().slice(0, 10) === date;
 }
+
+const filter = async (arr, predicate) =>
```

Comment:
consider naming `asyncFilter` when I first looked at the code I was thinking `await` was not needed. Adding async may make it clearer